website/docs/infrastructure/troubleshooting/appservice-hidden-appsettings.md

```diff
@@ -0,0 +1,139 @@
+or Terraform version. So we tried to replicate the issue in a safe environment
+by playing with the version of the two, by setting sensitive values in various
+ways: through a sensitive output, a sensitive variable or using the `sensitive`
+function.
+
+However, we couldn't reproduce the issue at all. We also found different
+behaviors in the same Terraform configuration(!).
+
+### GitHub issue on `azurerm` GitHub repository
+
+[We also opened an issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/28509)
+on `azurerm` GitHub repository, but without any luck yet (07/03/2025).
+
+## The Proposed Workaround
+
+By not being able to reproduce the issue, we propose the following workaround to
+mitigate the problem and restore developers confidence.
+
+The proposed workaround excludes any sensitive value from the app settings, by
+leveraging the native integration between AppService and KeyVault services.
+
+In fact, an AppService can access a KeyVault secret by:
+
+- Giving AppService system-assigned managed identity the access to read secrets:
+  - KeyVaults using Access Policies should give `Get` and `List` permissions
+  - KeyVaults using RBAC should be given `Key Vault Secrets User`
+- By referencing the secrets from the AppService environment variables
+  - Use the syntaxt
+    `@Microsoft.KeyVault(VaultName=<kv-name>;SecretName=<secret-name>)`
+
+Adopting this approach, Terraform code will not have any sensitive value to
+hide. Moreover, it is an Azure best practice with the following benefits:
+
+- Updating a value in KeyVault does not require a Terraform Apply anymore:
+  through the Azure Portal is possible to force new values pulling from KeyVault
+- If a secret reference is broken (e.g. missing secret, lack of read
+  permissions, etc.), Azure Portal highlights it in red
+- It becomes easier to track where secrets have been used
+
+### Techniques to Facilitate the Code Refactoring
+
+However, this transition require some code refactoring. We recommend to
+encapsulate the logic in the submodule of your AppService/Functions Apps in
+something like:
+
+```hcl
+locals {
+  local.function_apps.common_app_settings,
+  {
+    for s in var.app_settings :
+    s.name => s.key_vault_secret_name != null ? "@Microsoft.KeyVault(VaultName=${var.key_vault_name};SecretName=${s.key_vault_secret_name})" : s.value
+  }
+}
+
+variable "app_settings" {
+  type = list(object({
+    name                  = string
+    value                 = optional(string, "")
+    key_vault_secret_name = optional(string)
+  }))
+  description = "AppSettings configuration"
+}
+```
+
+Then, the caller root module could use this code:
+
+```hcl
+app_settings = [
+  {
+    name                  = "secret1"
+    key_vault_secret_name = "secret1"
+  },
+  {
+    name                  = "secret2"
+    key_vault_secret_name = "secret2"
+  },
+  {
+    name  = "plan1"
+    value = "plain value"
+  },
+]
+```
+
+### FAQs
```

Comment:
issue: avoid generic "FAQ" sections, try to put the content inline within the rest.

Reply:
see https://www.moghill.co.uk/blog/website-sins-3-why-you-dont-need-an-faq-page/

Reply:
https://github.com/pagopa/dx/pull/338/commits/8a08452798704902a9c0730d78e98cb4cb6c50e7